### ISSUE TITLE
Using "Set Start point" on Epilog Helix causes X-Axis limitation

### ISSUE THREAD

**Issue by zebulong** [t-oster/VisiCut#496]
I set my starting point (Red Cross hairs) on the work piece and started the cut, which is just a hole with smaller holes around it. the laser cut one hole, then when it went left, it looks like it hit a wall (but it did not). I thought my X-axis gear was loose. but then it went back to the right and cut the other smaller holes and when it went to the the big center hole, it was fine until it hit the imaginary wall like before and so now that part of the circle is just a straight line. I restarted and reloaded everything and it did it again. I removed the Red Cross-hairs starting point and ran a test and it cuts everything fine. it's when I set a start point that it does this a couple inches off to the left. any idea what could be doing this? I think it has to do with the start point. this is all taking place in the center of the Helix work bed, so there is plenty of room everywhere around it. I think this is a software bug. I have run it with and without start point and took a photo of the results. thanks for your help!
![20181229_182408](https://user-images.githubusercontent.com/36025857/50543849-6260f780-0b97-11e9-904e-09aee146bb2a.jpg)

**Comment by mgmax** [comment]
Yes, it's a software bug: A start point which isn't top-left of the drawing, but inside, is not correctly implemented in our Epilog driver. I'd guess you are the first one to ever use this combination.

We use negative coordinates in this case, however the driver does not change the coordinates but only uses special "offset" commands. I don't know exactly what these commands mean, more testing would be necessary to figure it out precisely.

Cut a 2x1 inch rectangle, which is placed with 3 inch margin to the left and 4 inch to the top. Print to file, look at the dump (I will upload my analysis tool soon, you can also use a hex editor. `\x1b` is the ASCII Escape character).

normal startpoint (top-left of page):
```
\x1b&y0Z
```

startpoint at center of rectangle (= 5 inch from the top, 3.5 inch from the left):
```
\x1b&y1Z (special startpoint?)
...
\x1b&l1750U (startpoint X?)
\x1b&l2495Z (startpoint Y?)
\x1b&l1995V (empty space above, can probably be set to zero?)
\x1b&l1500W (empty space left, can probably be set to zero?)
```


I currently have neither lots of time to work on this nor immediate access to an Epilog cutter, maybe someone else wants to take on this work? The only thing I will do is add an error message if this situation occurs.

**Comment by t-oster** [comment]
Strange... I remember that I successfully used the starting point feature with an Epilog Zing when I wrote the first versions of VisiCut.

**Comment by t-oster** [comment]
... although i only tested vector mode as far as I remember

**Comment by zebulong** [comment]
In the past, maybe a year ago, I think I did this successfully, but during the past months, I have run into this.  could it be that the bug occurred more recently since you first wrote it?

**Comment by mgmax** [comment]
At least on an Epilog Zing with newest firmware, I could not reproduce the bug. However the bug may occur with other models or firmware versions.

**Comment by t-oster** [comment]
So you are capable of using a startngpoint which is not top left and we were right to just add a warning in #96? Does it work in raster an vector mode?

**Comment by zebulong** [comment]
I will try next time I am layering and see what happens

I have a helix 18x24

**Comment by mgmax** [comment]
Yes, cutting works despite the warning. Unfortunately I did not test engraving.

**Comment by mgmax** [comment]
@zebulong did you find the time to try it?

**Comment by zebulong** [comment]
No my laser is broken and am waiting for parts
